services/src/main/scala/cromwell/pipeline/service/GitLabProjectVersioning.scala

```diff
@@ -30,22 +32,65 @@ class GitLabProjectVersioning(httpClient: HttpClient, config: GitLabConfig)
           resp =>
             if (resp.status != StatusCodes.Created.intValue)
               Left(VersioningException(s"The repository was not created. Response status: ${resp.status}"))
-            else Right(projectWithRepository(s"${config.idPath}${project.projectId.value}"))
+            else Right(projectWithRepository(config.idPath + project.projectId.value))
         )
         .recover { case e: Throwable => Left(VersioningException(e.getMessage)) }
     }
   }
+
   override def getFiles(project: Project, path: Path)(implicit ec: ExecutionContext): AsyncResult[List[String]] = ???
-  override def getProjectVersions(project: Project)(implicit ec: ExecutionContext): AsyncResult[Project] = ???
-  override def getFileVersions(project: Project, path: Path)(
-    implicit ec: ExecutionContext
-  ): AsyncResult[List[Version]] = ???
+
+  override def getProjectVersions(project: Project)(implicit ec: ExecutionContext): AsyncResult[Seq[Version]] = {
+    val versionsListUrl: String = s"${config.url}/projects/${project.repository}/repository/tags"
+    httpClient
+      .get(url = versionsListUrl, headers = config.token)
+      .map(
+        resp =>
+          if (resp.status != StatusCodes.OK.intValue)
+            Left(VersioningException(s"Could not take versions. Response status: ${resp.status}"))
+          else {
+            val versionsBody = Json.parse(resp.body).validate[Seq[Version]]
+            Right(versionsBody.get)
+          }
+      )
+      .recover { case e: Throwable => Left(VersioningException(e.getMessage)) }
+  }
```

Comment:
Why does your PR contain the changes from [PR](https://github.com/EpamLifeSciencesTeam/cmwl_pipeline/pull/63/files) made by @AndreiMaksimenko?

Reply:
Because his method used in my method. When I tried to find a endpoint for file versions, at one of the daily it was said, that we needed get all project versions and leave only those that are relevant to the desired file. Andrey suggested using his method, as its implementation is ready.

Reply:
This is strange :)
In that case, your PR can be merged only after Andrei's PR.

Reply:
I understand, It is supposed to be so.